pkg/controller/direct/sql/sqlinstance_controller.go

```diff
@@ -38,12 +39,330 @@ import (
+	settingsFieldPath                                       = "spec.settings"
+	editionFieldPath                                        = "spec.settings.edition"
+	activationPolicyFieldPath                               = "spec.settings.activationPolicy"
+	backupConfigurationFieldPath                            = "spec.settings.backupConfiguration"
+	backupRetentionSettingsFieldPath                        = "spec.settings.backupConfiguration.backupRetentionSettings"
+	backupRetentionSettingsRetainedBackupsFieldPath         = "spec.settings.backupConfiguration.backupRetentionSettings.retainedBackups"
+	backupRetentionSettingsRetentionUnitFieldPath           = "spec.settings.backupConfiguration.backupRetentionSettings.retentionUnit"
+	backupConfigurationEnabledFieldPath                     = "spec.settings.backupConfiguration.enabled"
+	backupConfigurationPointInTimeRecoveryEnabledFieldPath  = "spec.settings.backupConfiguration.pointInTimeRecoveryEnabled"
+	backupConfigurationStartTimeFieldPath                   = "spec.settings.backupConfiguration.startTime"
+	backupConfigurationTransactionLogRetentionDaysFieldPath = "spec.settings.backupConfiguration.transactionLogRetentionDays"
+	connectorEnforcementFieldPath                           = "spec.settings.connectorEnforcement"
+	dataCacheConfigFieldPath                                = "spec.settings.dataCacheConfig"
+	dataCacheConfigDataCacheEnabledFieldPath                = "spec.settings.dataCacheConfig.dataCacheEnabled"
+	diskAutoresizeFieldPath                                 = "spec.settings.diskAutoresize"
+	diskAutoresizeLimitFieldPath                            = "spec.settings.diskAutoresizeLimit"
+	diskSizeFieldPath                                       = "spec.settings.diskSize"
+	diskTypeFieldPath                                       = "spec.settings.diskType"
+	ipConfigurationFieldPath                                = "spec.settings.ipConfiguration"
+	ipConfigurationIPV4EnabledFieldPath                     = "spec.settings.ipConfiguration.ipv4Enabled"
+	locationPreferenceFieldPath                             = "spec.settings.locationPreference"
+	locationPreferenceZoneFieldPath                         = "spec.settings.locationPreference.zone"
+	pricingPlanFieldPath                                    = "spec.settings.pricingPlan"
 )
 
+// FieldMetadata encapsulates the state and logic for a field that can be unmanaged.
+type FieldMetadata struct {
+	// isUnmanaged tracks if the user has marked this field as unmanaged for a specific resource instance.
+	isUnmanaged bool
+
+	// preserveActualValue contains the specific logic to copy this field's value
+	// from the live GCP resource (`actual`) to the outgoing API request payload (`out`).
+
+	// if a field is nil in `out`, we treat it as a no op
+	preserveActualValue func(out *api.DatabaseInstance, actual *api.DatabaseInstance)
+}
+
+var supportedUnmanageableFields = map[string]*FieldMetadata{
+	instanceTypeFieldPath: {
```

Comment:
I personally think if a constant is single-use, it probably should be inline.  Because now I have to go check whether each constant lines up.